misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormattingTreeModifier.java

```diff
@@ -1391,7 +1391,7 @@ public TypeCastExpressionNode transform(TypeCastExpressionNode typeCastExpressio
     @Override
     public TypeCastParamNode transform(TypeCastParamNode typeCastParamNode) {
         NodeList<AnnotationNode> annotations = formatNodeList(typeCastParamNode.annotations(), 0, 1, 1, 0);
-        Node type = formatNode(typeCastParamNode.type(), env.trailingWS, env.trailingNL);
+        Node type = formatNode(typeCastParamNode.type().orElse(null), env.trailingWS, env.trailingNL);
```

Comment:
Formatting properties of the annotations changes depending on the existence of the `type`.
Can you create an issue to track this?

Reply:
Actually previously it was possible for the `typeCastParamNode.type()` to be null. However, it was handled inside the `formatNode` method accordingly.

```java
private <T extends Node> T formatNode(T node, int trailingWS, int trailingNL) {
        if (node == null || node.isMissing()) {
            return node;
        }
       // ...
}
```

Reply:
Trailing whitespaces/newlines etc. of a given node depends on the node to be followed. So here, if the `typeCastParamNode.type()` is not present, then the formatting the preceding node (metadata) must be adjusted accordingly.